src/librustc_mir/transform/const_prop.rs

```diff
@@ -820,7 +818,7 @@ impl<'tcx> Visitor<'tcx> for CanConstProp {
             | MutatingUse(MutatingUseContext::Borrow)
             | MutatingUse(MutatingUseContext::AddressOf) => {
                 trace!("local {:?} can't be propagaged because it's used: {:?}", local, context);
-                self.can_const_prop[local] = ConstPropMode::NoPropagation;
```

Comment:
@oli-obk so is this the diff that caused the bug? That these uses no longer prevent propagation? Instead of duplicating that use detection by handling `Ref` and `AddrOf` ourselves, would it make more sense to continue to use this info here (or at least do so additionally)?